hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,236 @@
+        */
+        DEVICE EvaluatorPairTable(Scalar _rsq, Scalar _rcutsq, const param_type& _params)
+            : rsq(_rsq), rcutsq(_rcutsq), width(_params.width), rmin(_params.rmin), V_table(_params.V_table), F_table(_params.F_table)
+            {
+            }
+
+        //! Table doesn't use diameter
+        DEVICE static bool needsDiameter() { return false; }
+        //! Accept the optional diameter values
+        /*! \param di Diameter of particle i
+            \param dj Diameter of particle j
+        */
+        DEVICE void setDiameter(Scalar di, Scalar dj) { }
+
+        //! Table doesn't use charge
+        DEVICE static bool needsCharge() { return false; }
+        //! Accept the optional diameter values
+        /*! \param qi Charge of particle i
+            \param qj Charge of particle j
+        */
+        DEVICE void setCharge(Scalar qi, Scalar qj) { }
+
+        //! Evaluate the force and energy
+        /*! \param force_divr Output parameter to write the computed force divided by r.
+            \param pair_eng Output parameter to write the computed pair energy
+            \param energy_shift If true, the potential must be shifted so that
+            V(r) is continuous at the cutoff
+            \note There is no need to check if rsq < rcutsq in this method.
+            Cutoff tests are performed in PotentialPair.
+
+            \return True if they are evaluated or false if they are not because
+            we are beyond the cutoff
+        */
+        DEVICE bool evalForceAndEnergy(Scalar& force_divr, Scalar& pair_eng, bool energy_shift)
+            {
+            Scalar rcut = fast::sqrt(rcutsq);
+            Scalar r = fast::sqrt(rsq);
+            // compute the force divided by r in force_divr
+            if (r < rcut && r >= rmin)
+                {
+                Scalar delta_r = (rcut - rmin) / width;
+                // precomputed term
+                Scalar value_f = (r - rmin) / delta_r;
+
+                // compute index into the table and read in values
+                unsigned int value_i = (unsigned int)floor(value_f);
+                // unpack the data
+                Scalar V0 = V_table[value_i];
+                Scalar V1 = 0;
+                Scalar F0 = F_table[value_i];
+                Scalar F1 = 0;
+                if (value_i + 1 < width)
+                    {
+                    V1 = V_table[value_i + 1];
+                    F1 = F_table[value_i + 1];
+                    }
+
+                // compute the linear interpolation coefficient
+                Scalar f = value_f - Scalar(value_i);
+
+                // interpolate to get V and F;
+                Scalar V = V0 + f * (V1 - V0);
+                Scalar F = F0 + f * (F1 - F0);
+
+                // convert to standard variables used by the other pair computes in HOOMD-blue
+                if (rsq > Scalar(0.0))
+                    force_divr = F / r;
+                pair_eng = Scalar(0.5) * V;
```

Comment:
Should this 0.5 be removed? The tests that compare the simulation forces to forces calculated by differentiating the simulation energy are failing by a factor of 1/2

Reply:
Yes, remove the 0.5. The code that calls the evaluator multiplies by 0.5 where necessary.